base/uk.ac.stfc.isis.ibex.ui.synoptic.editor/src/uk/ac/stfc/isis/ibex/ui/synoptic/editor/target/selector/ComponentNameVerifier.java

```diff
@@ -0,0 +1,45 @@
+ /*
+ * This file is part of the ISIS IBEX application.
+ * Copyright (C) 2012-2016 Science & Technology Facilities Council.
+ * All rights reserved.
+ *
+ * This program is distributed in the hope that it will be useful.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution.
+ * EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM 
+ * AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES 
+ * OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+ *
+ * You should have received a copy of the Eclipse Public License v1.0
+ * along with this program; if not, you can obtain a copy from
+ * https://www.eclipse.org/org/documents/epl-v10.php or 
+ * http://opensource.org/licenses/eclipse-1.0.php
+ */
+
+package uk.ac.stfc.isis.ibex.ui.synoptic.editor.target.selector;
+
+import org.eclipse.swt.events.VerifyEvent;
+import org.eclipse.swt.events.VerifyListener;
+
+/**
+ * Only allows text input if the new text is a valid component name.
+ */
+public class ComponentNameVerifier implements VerifyListener {
+    /**
+     * A regex matching a valid component name.
+     */
+    private static final String ALPHANUMERIC_SPACES_AND_UNDERSCORES_REGEX = "[a-zA-Z0-9_\\- ()]*";
```

Comment:
This is an input validator so it does need to match blank strings otherwise you can't delete the text from the input box (it actually stops you from typing in things that don't match this regex). But I agree there should be another check somewhere to check what you mentioned.